### PR TITLE
feat: add NodeController to support optional Closer and NodeUnwrapper

### PR DESCRIPTION
**Pull request by jimlambrt**

### CODE REVIEW

**Review by peteski22 — CHANGES_REQUESTED** [review]
Thanks Jim! 😄 I added a few bits, which hopefully make sense.

File: broker.go [hunk 1]
```diff
@@ -380,6 +381,10 @@ func (b *Broker) RemovePipelineAndNodes(t EventType, id PipelineID) error {
 
 		switch nodeUsage.referenceCount {
 		case 0, 1:
+			nc := NewNodeController(b.nodes[nodeID].node)
```
[the remark this answers is not in the document]
Comment (jimlambrt) [reply broker.go:384]: ty!

File: node.go [hunk 2]
```diff
@@ -32,6 +32,58 @@ type Node interface {
 	Type() NodeType
 }
 
+// A NodeController is used by the Broker to control the a node and may be used
+// by the broker whenever it needs to optionally control a node.  For instance,
+// if the broker wants to optionally ask a node to Close via the Closer
+// interface.
```
[the remark this answers is not in the document]
Comment (jimlambrt) [reply node.go:38]: ty!

File: node.go [hunk 3]
```diff
@@ -32,6 +32,58 @@ type Node interface {
...
+// should be the original value registered with the broker, or have an Unwrap
+// method returning the original Node (see NodeUnwrapper interface).
+//
+// If the Node implements any of the following methods, the NodeController will
+// call them as appropriate/needed:
+//
+//	Close()
+//
+// If the Node does not support a method, and the method signature includes an
+// error, then the NodeController returns an error matching ErrNotSupported.
```
[the remark this answers is not in the document]
Comment (jimlambrt) [reply node.go:51]: nope... ty!

File: broker.go [hunk 4]
```diff
@@ -380,6 +381,10 @@ func (b *Broker) RemovePipelineAndNodes(t EventType, id PipelineID) error {
...
+			if err := nc.Close(); err != nil {
+				return fmt.Errorf("unable to close node (ID == %q): %w", nodeID, err)
```
[the remark this answers is not in the document]
Comment (jimlambrt) [reply broker.go]: updated based on our slack discussion.

**Pull request merged**